Source/DafnyRuntime/DafnyRuntime.py

```diff
@@ -420,3 +420,35 @@ class defaults:
     real = staticmethod(BigRational)
     pointer = staticmethod(lambda: None)
     tuple = staticmethod(lambda *args: lambda: tuple(a() for a in args))
+
```

Comment:
I would prefer to keep the runtime as small as possible. Can we move those into a separate file in the `file-io` directory, until this becomes an actual language feature?

Reply:
@alex-chew and I revisited our reasoning for putting this logic in the runtime, and came to the same conclusion. The original motivation was piggybacking on the existing "distribution" of native code, but I think it's not worth the coupling cost, especially since it forces using refinement to customize the extern declaration per language.

@alex-chew is going to move this to a pure `dafny-lang/libraries` PR that includes the native code as well. The downside will be needing to manually include a target language artifact like `DafnyIO.jar` to build and run Dafny code that uses this library, but that pain should go away in the long term as we add better support for Dafny projects and dependencies, especially when you add separate compilation.

Reply:
Your call, but for me, including the IO routines in the runtime is well worth not having to scrounge around for the relevant FileIO.xx for each target. At least have `dafny run` include these extra files automatically.

Reply:
In the new PR there is no implementation in the runtime, so @fabiomadge's original concern in this thread is resolved.

> Your call, but for me, including the IO routines in the runtime is well worth not having to scrounge around for the relevant FileIO.xx for each target. At least have dafny run include these extra files automatically.

In my opinion the upsides of keeping the file I/O libraries independent from a particular Dafny version range, and of keeping the runtimes small, outweigh the cost of needing to specify an extra `--input libraries/src/FileIO/FileIO.xx` flag (or otherwise integrating the file into one's build process, if using separate compilation). But having `dafny run` include the extra files automatically is an interesting idea and I think it can be implemented later as a non-breaking change.